test/integration/project-management.spec.ts

```diff
@@ -175,6 +175,14 @@ describe('admin.projectManagement', () => {
             expect(certs.length).to.equal(0);
           });
     });
+
+    it('add a cert and then remove it fails', () => {
```

Comment:
This is testing the actual behaviour (both pre and post this PR) rather than the intended behaviour. It seems unfortunate that you can't delete a cert you just added (without going through getCerts first).

An alternative would be for the addShaCertificate to return a Promise<X> where X is either the resourceName or the cert itself, populated with the resourceName (rather than void.) But that requires a minor API change.

Another alternative would be to mutate the cert passed to addShaCertificate to populate the resourceName. But that's a bit awkward since (a) mutating parameters is always a little awkward, and (b) it may need to be done asynchronously, so there'd be a timing consideration there too.

Reply:
Returning a Promise<ShaCertificate> sounds like a good idea. And not a breaking change in this case.